librosa/sequence.py

```diff
@@ -359,6 +361,341 @@ def __dtw_backtracking(D_steps, step_sizes_sigma):  # pragma: no cover
     return wp
 
 
+def rqa(sim, gap_onset=1, gap_extend=1, knight_moves=True, backtrack=True):
+    '''Recurrence quantification analysis (RQA)
+
+    This function implements different forms of RQA as described by
+    Serra, Serra, and Andrzejak [1]_.  These methods take as input
+    a recurrence or cross-similarity matrix `sim`, and calculate the value
+    of path alignments by dynamic programming.
+
+    Note that unlike dynamic time warping (`dtw`), alignment paths here are
+    maximized, not minimized, so the input should measure similarity rather
+    than distance.
+
+    The simplest RQA method, denoted as `L` [1]_ (equation 3) and equivalent
+    to the method described by Eckman, Kamphorst, and Ruelle [2]_, accumulates
+    the length of diagonal paths with positive values in the input:
+
+        - `score[i, j] = score[i-1, j-1] + 1`  if `sim[i, j] > 0`
+        - `score[i, j] = 0` otherwise.
+
+    The second method, denoted as `S` [1]_ (equation 4), is similar to the first,
+    but allows for "knight moves" in addition to strict diagonal moves:
+
+        - `score[i, j] = max(score[i-1, j-1], score[i-2, j-1], score[i-1, j-2]) + 1`  if `sim[i, j] > 0`
+        - `score[i, j] = 0` otherwise.
+
+    The third method, denoted as `Q` [1]_ (equations 5 and 6) extends this by
+    allowing gaps in the alignment that incur some cost, rather than a hard
+    reset to 0 whenever `sim[i, j] == 0`.
+    Gaps are penalized by two additional parameters, `gap_onset` and `gap_extend`,
+    which are deducted from the value of the alignment path every time a gap
+    is introduced or extended (respectively).
+
+    Note that setting `gap_onset` and `gap_extend` to `np.inf` recovers the second
+    method, and disabling knight moves recovers the first.
+
+
+    .. [1] Serra, Joan, Xavier Serra, and Ralph G. Andrzejak.
```

Comment:
technically it's Joan Serrà and Xavier Serra, but i can see why you'd want to asciify Joan's last name into Serra

Reply:
We do set utf-8 encoding on all source files, so there's not really a problem here.  It may cause some difficulty in docstring rendering in REPL or sphinx, but I'll look into it.